### PR TITLE
New personabar Web Servers tab

### PR DESCRIPTION
**Pull request by donker**
Although the platform keeps track of web servers, there is no way to manage this data. This PR brings a web servers tab to the "Servers" personabar module as follows:

Screenshot 2021-01-09 at 22 47 20

You're able to edit the url for a web server and to delete them.

### CODE REVIEW

**Comment by mitchelsellers** [comment]
@donker thank you so much for the work on this, seeing this I'm curious if we can make a few small adjustments.

1.  Can we get the time of last activity displayed as well?  In situations where a server is brought online mid-day the last activity time will be critical.
2. Can we hide the "Remove inactive" if there are no servers in that is.
3. Can we add a server icon to the beginning of the row, and have it solid colored if active & greyed our if inactive?  (Thinking similar style to how we do things in the "Connectors" section

Items 2 & 3 are really nice to haves if at all possible.

**Comment by donker** [comment]
> 
> 
> @donker thank you so much for the work on this, seeing this I'm curious if we can make a few small adjustments.
> 
>     1. Can we get the time of last activity displayed as well?  In situations where a server is brought online mid-day the last activity time will be critical.
> 
>     2. Can we hide the "Remove inactive" if there are no servers in that is.
> 
>     3. Can we add a server icon to the beginning of the row, and have it solid colored if active & greyed our if inactive?  (Thinking similar style to how we do things in the "Connectors" section
> 
> 
> Items 2 & 3 are really nice to haves if at all possible.

Check, check, and check

**Comment by hismightiness** [comment]
🥇

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Servers/App_LocalResources/Servers.resx [hunk 1]
```diff
@@ -657,4 +657,64 @@ Are you sure you want to increment the version number for your site?</value>
...
+    <value>There was an error deleting this server</value>
+  </data>
+  <data name="Save.Text" xml:space="preserve">
+    <value>Save</value>
+  </data>
+  <data name="ServerUpdateUrlError.Text" xml:space="preserve">
+    <value>Couldn't update web server url</value>
+  </data>
+  <data name="Url.Help" xml:space="preserve">
+    <value>The url for this web server</value>
```
Comment (mitchelsellers) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Servers/App_LocalResources/Servers.resx:709]:
```suggestion
    <value>The URL for direct connection to this web server.  In multi-server environments, this is typically an IP address or unique DNS entry that points directly to an individual server.</value>
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Servers/App_LocalResources/Servers.resx [hunk 2]
```diff
@@ -657,4 +657,64 @@ Are you sure you want to increment the version number for your site?</value>
...
+  </data>
+  <data name="DeleteNonActiveServers.Confirm" xml:space="preserve">
+    <value>Do you wish to delete all inactive servers?</value>
+  </data>
+  <data name="DeleteNonActiveServers.Text" xml:space="preserve">
+    <value>Delete Non Active Servers</value>
+  </data>
+  <data name="DeleteNonActiveServers.Warning" xml:space="preserve">
+    <value>You can delete all servers that have not been active for a whole day below</value>
```
Comment (mitchelsellers) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Servers/App_LocalResources/Servers.resx:718]:
```suggestion
    <value>You can delete all servers that have not been active for 24 hours below</value>
```

**Review by mitchelsellers — DISMISSED** [review]
Overall this looks good, worked well with a big list of servers as well.  I did have two suggested changes for the help text, but still not 100% sure on the verbiage.

I would like @valadas or one of the others to review the React.Common changes for clarity.

**Review by valadas — APPROVED** [review]
I see nothing wrong in code and downloaded the CI build and it looks fine to me.
Thanks @donker

**Review by mtrutledge — APPROVED** [review]
This looks good. There is one spot where we reference a custom icon, but the icon exists in the SvgIcons we distribute. A new `type` should be added to the `getIcon` function of the IconButton Component. This would reserve the CustomIcon prop to components that use SVG icons that are not distributed with the Common library.

File: Dnn.AdminExperience/ClientSide/Roles.Web/src/components/roles/UsersInRole/UserRow/index.jsx [hunk 3]
```diff
@@ -110,7 +109,7 @@ class UserRow extends Component {
                 showIcon={true} showInput={false}
                 onIconClick={this.onExpiresTimeClick.bind(this, props.userDetails, props.index) }             />
         </span> : null;
-        let deleteAction = props.userDetails.allowDelete ? <IconButton type="x" width={17} onClick={this.onDeleteClick.bind(this, props.userDetails, props.index) } /> : null;
+        let deleteAction = props.userDetails.allowDelete ? <IconButton customIcon={require("!raw-loader!../../../../img/common/x.svg").default} width={17} onClick={this.onDeleteClick.bind(this, props.userDetails, props.index) } /> : null;
```
[the remark this answers is not in the document]
Comment (donker) [reply Dnn.AdminExperience/ClientSide/Roles.Web/src/components/roles/UsersInRole/UserRow/index.jsx:112]: I totally agree. We can do this in a separate PR for 9.9 as well if one has time.

**Pull request merged**